### PR TITLE
Update key-vault-configuration.md

### LINKED ISSUE #32690: Replace a few "Azure AD" to "Microsoft Entra ID"
### Description

[Enter feedback here]

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-8.0&branch=pr-en-us-32671

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/security/key-vault-configuration.md

### Document ID

efe5ae5e-ff64-f1c7-6920-4512fc8f36d1

### Article author

@Rick-Anderson

### PR DESCRIPTION
**Pull request by azarboon**
"Azure AD" is the old name so I've updated references to "Microsoft Entra ID"

Fixes #32690

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/security/key-vault-configuration.md](https://github.com/dotnet/AspNetCore.Docs/blob/d60c7a44e61e4fa41da3252214274c83981a6c8b/aspnetcore/security/key-vault-configuration.md) | [Azure Key Vault configuration provider in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?branch=pr-en-us-32671) |

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
We get this PR every month. There are some exceptions, I think this is an exception. If so, update the pr and in a comment `<!-- link to exception and state it's an exception -->`

File: aspnetcore/security/key-vault-configuration.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Add package references for the following packages:
...
-* `Managed`: Demonstrates how to use [Managed identities for Azure resources](/azure/active-directory/managed-identities-azure-resources/overview). The managed identity authenticates the app to Azure Key Vault with Azure Active Directory (AD) authentication without storing credentials in the app's code or configuration. The `Managed` version of the sample must be deployed to Azure. Follow the guidance in the [Use the managed identities for Azure resources](#use-managed-identities-for-azure-resources) section.
+* `Managed`: Demonstrates how to use [Managed identities for Azure resources](/azure/active-directory/managed-identities-azure-resources/overview). The managed identity authenticates the app to Azure Key Vault with Microsoft Entra ID authentication without storing credentials in the app's code or configuration. The `Managed` version of the sample must be deployed to Azure. Follow the guidance in the [Use the managed identities for Azure resources](#use-managed-identities-for-azure-resources) section.
```
Comment (Rick-Anderson) [inline aspnetcore/security/key-vault-configuration.md:32]:
```suggestion
* `Managed`: Demonstrates how to use [Managed identities for Azure resources](/azure/active-directory/managed-identities-azure-resources/overview). The managed identity authenticates the app to Azure Key Vault with Managed identities for Azure resources without storing credentials in the app's code or configuration. The `Managed` version of the sample must be deployed to Azure. Follow the guidance in the [Use the managed identities for Azure resources](#use-managed-identities-for-azure-resources) section.
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/security/key-vault-configuration.md [hunk 2]
```diff
@@ -402,7 +402,7 @@ Add package references for the following packages:
...
+* `Managed`: Demonstrates how to use [Managed identities for Azure resources](/azure/active-directory/managed-identities-azure-resources/overview). The managed identity authenticates the app to Azure Key Vault with Microsoft Entra ID authentication without credentials stored in the app's code or configuration. When using managed identities to authenticate, an Microsoft Entra ID Application ID and Password (Client Secret) aren't required. The `Managed` version of the sample must be deployed to Azure. Follow the guidance in the [Use the managed identities for Azure resources](#use-managed-identities-for-azure-resources) section.
```
Comment (Rick-Anderson) [inline aspnetcore/security/key-vault-configuration.md:405]:
```suggestion
* `Managed`: Demonstrates how to use [Managed identities for Azure resources](/azure/active-directory/managed-identities-azure-resources/overview). The managed identity authenticates the app to Azure Key Vault with Managed identities for Azure resources without credentials stored in the app's code or configuration. When using managed identities to authenticate, a Managed identities for Azure resources app ID and Password (Client Secret) aren't required. The `Managed` version of the sample must be deployed to Azure. Follow the guidance in the [Use the managed identities for Azure resources](#use-managed-identities-for-azure-resources) section.
```

File: aspnetcore/security/key-vault-configuration.md [hunk 3]
```diff
@@ -103,7 +103,7 @@ Complete the following steps to create an Azure Key Vault and store the sample a
 
 ## Use Application ID and X.509 certificate for non-Azure-hosted apps
 
-Configure Azure AD, Azure Key Vault, and the app to use an Azure AD Application ID and X.509 certificate to authenticate to a vault **when the app is hosted outside of Azure**. For more information, see [About keys, secrets, and certificates](/azure/key-vault/about-keys-secrets-and-certificates).
+Configure Microsoft Entra ID, Azure Key Vault, and the app to use an Microsoft Entra ID Application ID and X.509 certificate to authenticate to a vault **when the app is hosted outside of Azure**. For more information, see [About keys, secrets, and certificates](/azure/key-vault/about-keys-secrets-and-certificates).
```
Comment (Rick-Anderson) [inline aspnetcore/security/key-vault-configuration.md:106]:
```suggestion
Configure Azure Key Vault and the app to use an Microsoft Entra ID Application ID and X.509 certificate to authenticate to a vault **when the app is hosted outside of Azure**. For more information, see [About keys, secrets, and certificates](/azure/key-vault/about-keys-secrets-and-certificates).
```

File: aspnetcore/security/key-vault-configuration.md [hunk 4]
```diff
@@ -476,7 +476,7 @@ Complete the following steps to create an Azure Key Vault and store the sample a
...
```
Comment (Rick-Anderson) [inline aspnetcore/security/key-vault-configuration.md:479]:
```suggestion
Configure Azure Key Vault and the app to use an Microsoft Entra ID Application ID and X.509 certificate to authenticate to a vault **when the app is hosted outside of Azure**. For more information, see [About keys, secrets, and certificates](/azure/key-vault/about-keys-secrets-and-certificates).
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

**Review by Rick-Anderson — APPROVED** [review]

**Comment by azarboon** [comment]
@dotnet-policy-service agree

**Pull request merged**